webapp/src/main/java/org/apache/atlas/web/rest/TypesREST.java

```diff
@@ -574,6 +574,16 @@ private void validateBuiltInTypeNames(AtlasTypesDef typesDef) throws AtlasBaseEx
                 if (typeDefStore.hasBuiltInTypeName(typeDef))
                     throw new AtlasBaseException(AtlasErrorCode.FORBIDDEN_TYPENAME, typeDef.getName());
         }
+        if (CollectionUtils.isNotEmpty(typesDef.getBusinessMetadataDefs())) {
+            for (AtlasBaseTypeDef typeDef : typesDef.getBusinessMetadataDefs())
+                if (typeDefStore.hasBuiltInTypeName(typeDef))
+                    throw new AtlasBaseException(AtlasErrorCode.FORBIDDEN_TYPENAME, typeDef.getName());
+        }
+        if (CollectionUtils.isNotEmpty(typesDef.getClassificationDefs())) {
```

Comment:
What I am saying is that this check right here won't tell anything. This is basically to check we are not passing any built-in type. So this will not check that if the type was invalid or not. So that error is coming from somewhere else. Most probably coming from : https://github.com/atlanhq/atlas-metastore/blob/b1b719696099dfe48a746d96d308f36bd5858846/repository/src/main/java/org/apache/atlas/repository/store/graph/v2/AtlasBusinessMetadataDefStoreV2.java#L315